packages/types/test/unit/ExecutionAddress.test.ts

```diff
@@ -0,0 +1,61 @@
+import {describe, it, expect} from "vitest";
+import {toChecksumAddress} from "../../src/utils/ExecutionAddress.js";
+
+describe("toChecksumAddress", () => {
+  it("should fail with invalid addresses", () => {
+    expect(() => toChecksumAddress("0x1234")).toThrowError("Invalid address");
+  });
+
+  it("should format address as ERC55", () => {
+    type TestCase = {
+      address: string;
+      checksumedAddress: string;
+    };
+
+    expect(toChecksumAddress("52908400098527886E0F7030069857D2E4169EE7")).toBe(
+      "0x52908400098527886E0F7030069857D2E4169EE7"
+    );
+
+    const testCases: TestCase[] = [
+      // All caps
+      {
+        address: "0x52908400098527886E0F7030069857D2E4169EE7",
+        checksumedAddress: "0x52908400098527886E0F7030069857D2E4169EE7",
+      },
+      {
+        address: "0x8617E340B3D01FA5F11F306F4090FD50E238070D",
+        checksumedAddress: "0x8617E340B3D01FA5F11F306F4090FD50E238070D",
+      },
+      // All lower
+      {
+        address: "0xde709f2102306220921060314715629080e2fb77",
+        checksumedAddress: "0xdE709F2102306220921060314715629080e2fB77",
+      },
+      {
+        address: "0x27b1fdb04752bbc536007a920d24acb045561c26",
+        checksumedAddress: "0x27B1FDb04752BBC536007A920d24acB045561C26",
+      },
+      // Normal
+      {
+        address: "0x5aAeb6053F3E94C9b9A09f33669435E7Ef1BeAed",
+        checksumedAddress: "0x5AAEB6053F3E94C9b9A09F33669435E7Ef1BeAED",
```

Comment:
How is this test case passing? `0x5AAEB6053F3E94C9b9A09F33669435E7Ef1BeAED` is definitely not a valid checksum address

see https://www.eip55.xyz/, or https://etherscan.io/ which also converts to checksum addresses

Reply:
Good thing that you double checked, as the implementation was broken and I misunderstand the set of test cases from the EIP.